icu4c/source/data/misc/supplementalData.txt

```diff
@@ -35597,398 +35597,297 @@ supplementalData:table(nofallback){
-        }
-        MY{
-            concentr-blood-glucose{"millimole-per-liter"}
-            consumption-vehicle-fuel{"liter-per-kilometer"}
-            length-person{"meter centimeter"}
-        }
-        MZ{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        NA{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        NL{
-            concentr-blood-glucose{"millimole-per-liter"}
-            consumption-vehicle-fuel-informal{"liter-per-kilometer"}
-            length-person-informal{"meter centimeter"}
-            length-visiblty{"meter"}
-        }
-        NO{
-            concentr-blood-glucose{"millimole-per-liter"}
-            length-person-informal{"meter centimeter"}
-            speed-wind{"meter-per-second"}
-        }
-        NZ{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        OM{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        PG{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        PL{
-            length-person-informal{"meter centimeter"}
-            speed-wind{"meter-per-second"}
-        }
-        PR{
-            temperature-weather{"fahrenheit"}
-        }
-        PT{
-            length-person-informal{"meter centimeter"}
-        }
-        PW{
-            temperature-weather{"fahrenheit"}
-        }
-        RS{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        RU{
-            concentr-blood-glucose{"millimole-per-liter"}
-            length-person-informal{"meter centimeter"}
-            pressure-baromtrc{"millimeter-of-mercury"}
-            speed-wind{"meter-per-second"}
-        }
-        SA{
-            length-person{"meter centimeter"}
-            mass-person-small{"kilogram gram"}
-        }
-        SE{
-            concentr-blood-glucose{"millimole-per-liter"}
-            length-person{"meter centimeter"}
-            length-road-informal{"mile-scandinavian"}
-            speed-wind{"meter-per-second"}
-        }
-        SG{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        SI{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        SK{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        TC{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        TH{
-            consumption-vehicle-fuel{"liter-per-kilometer"}
-            pressure-baromtrc{"millibar"}
-        }
-        TO{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        TR{
-            consumption-vehicle-fuel{"liter-per-kilometer"}
-            length-person{"meter centimeter"}
-        }
-        UA{
-            concentr-blood-glucose{"millimole-per-liter"}
-        }
-        UG{
-            concentr-blood-glucose{"millimole-per-liter"}
+        "area"{
+            "geograph"{
+                001{
+                    square-kilometer{""}
```

Comment:
The data types should be the same.  Here, the unit is a string, but below, it is a table.  You should pick one, probably table.  You can declare this as an empty table by appending `:table`:

```
square-kilometer:table{}
```

Reply:
From a discussion with hagbard it sounds like paths have to be predictable (can't be iterated over unless we use an :array), so unit identifiers need to move back into values.

I've reworked this sample, what do you think of the enumerated approach?